sklearn/model_selection/_subselect.py

```diff
@@ -0,0 +1,573 @@
+        # columns against the best average score column
+        tests = [
+            pair
+            for pair in list(itertools.combinations(range(score_grid.shape[0]), 2))
+            if best_score_idx in pair
+        ]
+
+        pvals = {}
+        for pair in tests:
+            pvals[pair] = wilcoxon(
+                score_grid[pair[0]],
+                score_grid[pair[1]],
+                alternative=self.alternative,
+                zero_method=self.zero_method,
+            )[1]
+
+        # Return the models that are insignificantly different from the best average
+        # performing, else just return the best-performing model.
+        surviving_ranks = [pair[0] for pair in tests if pvals[pair] > self.alpha] + [
+            best_score_idx
+        ]
+
+        if len(surviving_ranks) == 0:
+            surviving_ranks = [best_score_idx]
+            warnings.warn(
+                "The average performance of all cross-validated models is "
+                "significantly different from that of the best-performing model.",
+                UserWarning,
+            )
+
+        h_cutoff = np.nanmax(cv_means[surviving_ranks])
+        l_cutoff = np.nanmin(cv_means[surviving_ranks])
+
+        return l_cutoff, h_cutoff
+
+
+class by_fixed_window:
+    """A callable class that returns a fixed window of model performance based on
+    user-specified bounds.
+    """
+
+    def __init__(
+        self, lower_bound: Optional[float] = None, upper_bound: Optional[float] = None
+    ):
+        self.lower_bound = lower_bound
+        self.upper_bound = upper_bound
+
+    def __call__(
+        self,
+        score_grid: np.ndarray,
+        cv_means: np.ndarray,
+        best_score_idx: int,
+        n_folds: int,
+    ) -> Tuple[float, float]:
+        l_cutoff, h_cutoff = self.lower_bound, self.upper_bound
+        if l_cutoff is None:
+            l_cutoff = np.nanmin(cv_means)
+        if h_cutoff is None:
+            h_cutoff = np.nanmax(cv_means)
+
+        return l_cutoff, h_cutoff
+
+
+class Refitter:
+    """`Refitter` is an interface that generates any of a variety of tuning refit
+    callables for use with `GridSearchCV` and `RandomSearchCV`. A refit callable can be
+    used to select a best-performing model from among a constrained subset of models.
+    This can be useful for instance the case that the user wishes to identify
+    best-performing models above or below a particular threshold. It can also be useful
+    for selecting alternative models whose performance is not meaningfully different
+    from the best-performing model, but whose simplicity may be more preferable (e.g.
+    to prevent overfitting).
+
+    Attributes
+    ----------
+    cv_results_ : dict of numpy (masked) ndarrays
+        A dict with keys as column headers and values as columns, that can be
+        imported into a pandas ``DataFrame``.
+
+        For instance the below given table
```

Comment:
No need to repeat all this. Just say "see GridSearchCV" or similar

Reply:
Cool, removed.